packages/api-plugin-carts/src/util/publishCartUpdatedEvent.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @summary Publishes a cart updated event
+ * @param {Object} context - The application context
+ * @param {Object} cart - The cart that was updated
+ * @param {Boolean} params.publishUpdatedEvent - Whether to prevent publishing the event
+ * @returns {void} - undefined
+ */
+export default function publishCartUpdatedEvent(context, cart, { publishUpdatedEvent = undefined }) {
+  if (!context.app.hasSubscriptionsEnabled || !publishUpdatedEvent) return;
+  if (cart.accountId) {
+    context.pubSub.publish("ACCOUNT_CART_UPDATED", { accountCartUpdate: cart });
+  } else {
+    context.pubSub.publish("ANONYMOUS_CART_UPDATED", { anonymousCartUpdate: cart });
```

Comment:
It seems fine for both of these to pass `{ cart }` rather than separate names `accountCartUpdate` and `anonymousCartUpdate`. The subscriptions could also then share the same `filter` function if you want.

Reply:
You might even consider simplifying this to emit `CART_UPDATED` always, and then filter on `cart.accountId` in each subscription resolver instead. There are probably pros and cons to each approach, performance vs. consumer complexity.

Reply:
@vanpho93 I still think this should be simplified further to:

```js
context.pubSub.publish("CART_UPDATED", { cart });
```

Otherwise it is unnecessary complexity for the event consumer to check for both `anonymousCartUpdate` and `accountCartUpdate` when they often will not care what type of cart it is. If they do care, they can check it with `if (cart.accountId)` in the consumer function.

For example, if you make this change, then your account cart subscription filter can just be:

```js
export function filter(payload, variables) {
  const { cart } = payload;
  const { accountId, shopId } = variables;
  return cart.accountId === accountId && cart.shopId === shopId;
}
```